bpf/lib/lb.h

```diff
@@ -1643,7 +1643,7 @@ static __always_inline int lb4_local(const void *map, struct __ctx_buff *ctx,
 			if (backend && !state->syn)
 				break;
 
-			svc = lb4_lookup_service(key, false, true);
+			svc = lb4_lookup_service(key, is_defined(ENABLE_NODEPORT), true);
```

Comment:
As suggested in https://github.com/cilium/cilium/issues/32472#issuecomment-2117699560, please make this a parameter. So that the [call from nodeport.h](https://github.com/cilium/cilium/blob/963c40671c529f5fe8d80b20b778e2b79e7e3f3e/bpf/lib/nodeport.h#L2840) can [consistently](https://github.com/cilium/cilium/blob/963c40671c529f5fe8d80b20b778e2b79e7e3f3e/bpf/lib/nodeport.h#L2813) pass `false` for the `scope_switch`.

Reply:
sorry, misinterpreted the suggestion. Done